diode-server/reconciler/logs_retriever.go

```diff
@@ -237,5 +218,43 @@ func retrieveIngestionLogs(ctx context.Context, logger *slog.Logger, client Redi
 		metrics.Total = response.TotalResults
 	}
 
-	return &reconcilerpb.RetrieveIngestionLogsResponse{Logs: logs, Metrics: &metrics, NextPageToken: encodeInt64ToBase64(ingestionTs)}, nil
+	return &reconcilerpb.RetrieveIngestionLogsResponse{Logs: logs, Metrics: &metrics, NextPageToken: nextPageToken}, nil
+}
+
+func buildQueryFilter(req *reconcilerpb.RetrieveIngestionLogsRequest) string {
+	queryFilter := "*"
+
+	// apply optional filters for ingestion timestamps (start and end)
+	if req.GetIngestionTsStart() > 0 || req.GetIngestionTsEnd() > 0 {
+		ingestionTsFilter := fmt.Sprintf("@ingestion_ts:[%d inf]", req.GetIngestionTsStart())
+
+		if req.GetIngestionTsEnd() > 0 {
```

Comment:
looking good! it was missing end TS due to the way next page was being performed. With that change it makes sense